rbi/core/dir.rbi

```diff
@@ -567,18 +567,10 @@ class Dir < Object
   # Equivalent to calling `Dir.glob([string,...], 0)`.
   sig do
     params(
-        pattern: T.any(String, T::Array[String]),
-        flags: Integer,
+        pattern: String,
+        base: T.nilable(String)
```

Comment:
Is this the correct way to indicate that `base` is optional?

Reply:
I just realized that `base` can also be a `Pathname` too.

Reply:
I've allowed both `pattern` and `base` to be `String` or `Pathname` in the latest changes.